File: solved/1107리모컨.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>
#include <string>
#define START 100

using namespace std;


int main() {
    int N, M;
    cin >> N >> M;

    vector<bool> isDisable(10, false);

    int a;
    for(int i = 0; i < M; i++) {
        cin >> a;
        isDisable[a] = true;
    }


    int minVal = 1000000;
    for(int i = 0; i <= 1000000; i++) {
        int term = abs(N - START);

        bool canMake = true;
        string str = to_string(i);

        for(int j = 0; j < str.size(); j++) {
            if(isDisable[str[j] - '0']) {
                canMake = false;
                break;
            }
        }

        if(canMake) {
            term = min(term, (int)str.size() + abs(N - i));
        }

        if(term < minVal) {
            minVal = term;
        }

    }

    cout << minVal;

    return 0;
}

File: solved/1107리모컨.py
N = int(input())
M = int(input())
list_disable = []
if M > 0:
    list_disable = map(int, input().split())
is_enable = [True] * 10

for i, x in enumerate(list_disable):
    is_enable[x] = False

min_val = float('inf')
for num in range(1000001):
    term = abs(N - 100)

    can_make_flag = True
    s = str(num)
    for c in s:
        if not is_enable[ord(c) - ord('0')]:
            can_make_flag = False
            break
    
    if can_make_flag:
        term = min(term, len(s) + abs(N - num))
    
    if term < min_val:
        min_val = term

print(min_val)